### PR TITLE
Support wildcards and meta items in diff/commit filters

### PR DESCRIPTION
**Pull request by craigds**
## Description

This PR adds support for the following filter styles:

* `kart diff HEAD^ datasetname:meta:title` - diff a meta item
* `kart diff HEAD^ datasetname:meta` - diff all meta items from datasetname
* `kart diff HEAD^ *:meta` - diff all meta items from all datasets
* `kart diff HEAD^ *:feature` - diff all features from all datasets
* `kart diff HEAD^ *abc*:feature` - diff all features from datasets whose names contain `abc`

In addition specific-feature filters no longer also show meta items.

### Todo:

- ~~[ ] Prevent user from committing a feature without an associated schema change (`kart commit '*:feature'` when a schema change is also present, as mentioned in #219 )~~
- [x] implement or disable wildcard filters for `log` (it's hard)

## Related links:

* #219 
* #514

## Checklist:

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Review by olsen232 — APPROVED** [review]
Nice work!

**Comment by craigds** [comment]
>  Prevent user from committing a feature without an associated schema change (kart commit '*:feature' when a schema change is also present

This in fact turns out to not be a problem since Kart actually commits your meta items whenever you commit any feature. This might be improved in future to be less implicit, but for now it seems okay and doesn't need to explicitly prevent you from committing a feature when you have a local schema change.

**Comment by craigds** [comment]
I disabled the specific case of wildcard dataset name with feature IDs (e.g. `*:feature:123`), because it's quite tricky to implement. Since `kart log` delegates to `git log`, supporting this would mean we would need to figure out the full paths of features affected by the pathspecs in advance of calling git log. But we'd need to look at the associated schemas for those features to figure out the mapping from PK to feature path, and the mapping may change over time...

In short, it's very difficult unless we stop calling `git log` and implement log ourselves, so I just disabled it:

```
$ kart log -- *:feature:0008240571
Error: `kart log` doesn't currently support filters with both wildcards and feature IDs
```

I don't expect anyone will notice for a good while, in which time maybe we'll implement log ourselves

**Review by olsen232 — APPROVED** [review]
